### PR TITLE
feat: Apply Resource Validator on SAM GraphQL Resources

### PR DESCRIPTION
**Pull request by GavinZZ**
### Description of changes
Use resource validator as a runtime check on nested properties. No need to use `sam_expect` anymore.

### Description of how you validated changes
All tests passed

### Checklist

- [ ] Adheres to the [development guidelines](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#development-guidelines)
- [ ] Add/update [transform tests](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#unit-testing-with-multiple-python-versions)
    - [ ] Using correct values
    - [ ] Using wrong values
- [ ] Add/update [integration tests](https://github.com/aws/serverless-application-model/blob/develop/INTEGRATION_TESTS.md)

### Examples?

Please reach out in the comments if you want to add an example. Examples will be 
added to `sam init` through [aws/aws-sam-cli-app-templates](https://github.com/aws/aws-sam-cli-app-templates).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: samtranslator/internal/model/appsync.py [hunk 1]
```diff
@@ -18,15 +18,15 @@ class DeltaSyncConfigType(TypedDict):
...
+    AwsRegion: Union[str, Dict[str, str]]
     TableName: str
     UseCallerCredentials: bool
     Versioned: bool
     DeltaSyncConfig: DeltaSyncConfigType
 
 
 class LogConfigType(TypedDict, total=False):
-    CloudWatchLogsRoleArn: Intrinsicable[str]
+    CloudWatchLogsRoleArn: Optional[Intrinsicable[str]]
```
Comment (pradhapanr) [inline samtranslator/internal/model/appsync.py:29]: [Optional doesn't work in TypedDict](https://stackoverflow.com/questions/68814891/typeddict-does-not-allow-optional-keys)

File: tests/translator/output/error_graphqldatasource.json [hunk 2]
```diff
@@ -1,3 +1,3 @@
 {
-  "errorMessage": "Invalid Serverless Application Specification document. Number of errors found: 8. Resource with id [NoApiId] is invalid. Missing required property 'ApiId'. Resource with id [NoBaseTableTTL] is invalid. Property 'DynamoDBConfig.DeltaSyncConfig.BaseTableTTL' should be a string. Resource with id [NoConfigPropertyDefined] is invalid. 'DynamoDBConfig' must be defined when 'Type' is 'AMAZON_DYNAMODB'. Resource with id [NoDeltaSyncTableName] is invalid. Property 'DynamoDBConfig.DeltaSyncConfig.DeltaSyncTableName' should be a string. Resource with id [NoDeltaSyncTableTTL] is invalid. Property 'DynamoDBConfig.DeltaSyncConfig.DeltaSyncTableTTL' should be a string. Resource with id [NoDynamoDBTableName] is invalid. Property 'DynamoDBConfig.TableName' should be a string. Resource with id [NoType] is invalid. Missing required property 'Type'. Resource with id [UnknownType] is invalid. 'WEIRD_TYPE' is not a supported data source type."
```
Comment (hoffa) [inline tests/translator/output/error_graphqldatasource.json:2]: Can you add a test that adding some unknown property under `Logging` causes failure? 👉👈
So far I don't think we've failed, but this hopefully will (which is great).

File: samtranslator/internal/schema_source/aws_serverless_graphqlapi.py [hunk 3]
```diff
@@ -13,7 +19,7 @@ class Auth(BaseModel):
 
 
 class Logging(BaseModel):
-    CloudWatchLogsRoleArn: Optional[str]
+    CloudWatchLogsRoleArn: Optional[SamIntrinsicable[str]]
```
Comment (pradhapanr) [inline samtranslator/internal/schema_source/aws_serverless_graphqlapi.py:22]: @hoffa @aaythapa Are we still supposed to be marking fields as intrinsicable on SAM models?

File: samtranslator/model/sam_resources.py [hunk 4]
```diff
@@ -2193,18 +2203,20 @@ def _create_logging_default(self) -> Tuple[LogConfigType, IAMRole]:
...
-            log_config["ExcludeVerboseContent"] = self.Logging["ExcludeVerboseContent"]
+        if model.Logging.ExcludeVerboseContent:
+            log_config["ExcludeVerboseContent"] = cast(
+                bool, model.Logging.ExcludeVerboseContent
+            )  # TODO: better handling PassThroughProp
 
-        log_config["FieldLogLevel"] = self.Logging.get("FieldLogLevel", "ALL")
-        log_config["CloudWatchLogsRoleArn"] = self.Logging.get("CloudWatchLogsRoleArn", None)
+        log_config["FieldLogLevel"] = model.Logging.FieldLogLevel or "ALL"
+        log_config["CloudWatchLogsRoleArn"] = model.Logging.CloudWatchLogsRoleArn
```
Comment (pradhapanr) [inline samtranslator/model/sam_resources.py:2215]: So if `CloudWatchLogsRoleArn` isn't defined in template, then in `model.Logging.CloudWatchLogsRoleArn` is set to `None`?

**Review by pradhapanr — COMMENTED** [review]
These are going to be very useful!! Thanks for taking it on.

Left small comments, but looks good overall. I'm creating our `DataSource` changes now so this will be very helpful in translating that over using models.

File: samtranslator/internal/schema_source/aws_serverless_graphqlapi.py (hunk 3 above)
Comment (pradhapanr) [reply samtranslator/internal/schema_source/aws_serverless_graphqlapi.py:22]: Sure they could be marked as passthrough then according to that definition. I was marking properties as `PassThrough` if there was no interaction with the value in the transform code outside of just assigning it to the CFN resource

File: samtranslator/model/sam_resources.py (hunk 4 above)
Comment (GavinZZ) [reply samtranslator/model/sam_resources.py:2215]: `CloudWatchLogsRoleArn=None ExcludeVerboseContent=PassThroughProp(__root__=True) FieldLogLevel='ERROR'`

File: samtranslator/internal/schema_source/aws_serverless_graphqlapi.py (hunk 3 above)
Comment (GavinZZ) [reply samtranslator/internal/schema_source/aws_serverless_graphqlapi.py:22]: Updated to PassThroughProp.

File: samtranslator/internal/model/appsync.py (hunk 1 above)
Comment (GavinZZ) [reply samtranslator/internal/model/appsync.py:29]: Good call. Let me remove it.

File: tests/translator/output/error_graphqldatasource.json (hunk 2 above)
Comment (GavinZZ) [reply tests/translator/output/error_graphqldatasource.json:2]: Added

**Review by hoffa — APPROVED** [review]

**Comment by xazhao** [comment]
The error messages are kinda hard to review. Can we integrate with [this change ](https://github.com/aws/serverless-application-model/pull/2968)so it's easier to review?

**Comment by GavinZZ** [comment]
> The error messages are kinda hard to review. Can we integrate with [this change ](https://github.com/aws/serverless-application-model/pull/2968)so it's easier to review?

Going to do that in a separate PR. This branch `feat-appsync` is super outdated compared to develop.

**Review by xazhao — APPROVED** [review]

**Review by aahung — APPROVED** [review]

**Pull request merged**